internal/codecs/array.go

```diff
@@ -45,17 +45,33 @@ type Array struct {
 	id    types.UUID
 	child Codec
 	typ   reflect.Type
-	step  int
+
+	// step is the element width in bytes for a go array of type `Array.typ`.
+	step int
```

Comment:
Why is this needed btw? I don't recall having this field in Python or JS implementations.

Reply:
This is needed because decoding is done by writing data to known memory locations based on the out value's type. When there is a slice (an array under the hood) in the data structure the codec needs to know where each array element starts in memory. Something like:

```go
var head unsafe.Pointer

for i := 0; i < elemCount; i++ {
    decodeNextElementToLocation(head + i*step)
}
```